### PR TITLE
Update error messages to be more specific

### LINKED ISSUE #692: Add more verbose error when camera not found
Right now we just look in NT for the expected keys and, if not found, just report an error. This causes the same symptom for two different root cause classes: Possibly, the camera is not connected or running at all. Or, possibly, the user code and camera UI names are not aligned.

A better way might be able to list out what cameras _were_ seen (IE, query network tables for the keys). This would better differentiate the root causes.

### PR DESCRIPTION
**Pull request by hadley31**
Closes #692

### CODE REVIEW

**Review by mcm001 — APPROVED** [review]

File: photon-lib/src/main/java/org/photonvision/PhotonCamera.java [hunk 1]
```diff
@@ -292,12 +295,25 @@ private void verifyVersion() {
         // Heartbeat entry is assumed to always be present. If it's not present, we
         // assume that a camera with that name was never connected in the first place.
         if (!heartbeatEntry.exists()) {
```
Comment (mcm001) [inline photon-lib/src/main/java/org/photonvision/PhotonCamera.java:297]: Actually I can't get this if to be true in my simulation

File: photon-lib/src/main/native/cpp/photonlib/PhotonCamera.cpp [hunk 2]
```diff
@@ -134,10 +134,29 @@ void PhotonCamera::VerifyVersion() {
   const std::string& versionString = versionEntry.Get("");
   if (versionString.empty()) {
     std::string path_ = path;
-    FRC_ReportError(
-        frc::warn::Warning,
-        "PhotonVision coprocessor at path {} not found on NetworkTables!",
-        path_);
+    std::vector<std::string> cameraNames =
```
Comment (mcm001) [inline photon-lib/src/main/native/cpp/photonlib/PhotonCamera.cpp:137]: Actually. this doesn't work locally for me. You'll only get back a vector of subtables your client has subscribed to, it seems like? 
image

**Review by mcm001 — CHANGES_REQUESTED** [review]
Seems like getsubtables doesn't return every subtable the server is serving? At least not in my testing

File: photon-lib/src/main/native/cpp/photonlib/PhotonCamera.cpp (hunk 2 above)
Comment (mcm001) [reply photon-lib/src/main/native/cpp/photonlib/PhotonCamera.cpp:137]: MVP shown here: https://cdn.discordapp.com/attachments/368993897495527424/1060342579381669978/image.png

File: photon-lib/src/main/java/org/photonvision/PhotonCamera.java [hunk 3]
```diff
@@ -106,7 +109,7 @@ public static void setVersionCheckEnabled(boolean enabled) {
      */
     public PhotonCamera(NetworkTableInstance instance, String cameraName) {
```
Comment (mcm001) [inline photon-lib/src/main/java/org/photonvision/PhotonCamera.java:110]: Fix for the below: add a MultiSubscriber member, and add this to the constructor:

```java
        multiSubscriber = new MultiSubscriber(instance, 
            new String[]{"/photonvision/"},
            new PubSubOption[]{PubSubOption.topicsOnly(true)});
```

**Review by mcm001 — CHANGES_REQUESTED** [review]

File: photon-lib/src/main/native/cpp/photonlib/PhotonCamera.cpp (hunk 2 above)
Comment (hadley31) [reply photon-lib/src/main/native/cpp/photonlib/PhotonCamera.cpp:137]: @mcm001 Based on the screenshot, is it possible that the `photonvision` table wasn't initialized yet when you made those calls?

Comment (gerth2) [reply photon-lib/src/main/native/cpp/photonlib/PhotonCamera.cpp:137]: Correct, this is an NT4ism.

Our implementation should first subscribe to `/photonvision/*` with the "topic names only" flag set. Then downstream things should work.

**Review by mcm001 — APPROVED** [review]

**Review by mcm001 — APPROVED** [review]

**Pull request merged**